core/src/main/scala/spinal/core/formal/FormalBootstraps.scala

```diff
@@ -23,8 +23,10 @@ package spinal.core.formal
 import java.io.{File, PrintWriter}
 import org.apache.commons.io.FileUtils
 import spinal.core.internals.{PhaseContext, PhaseNetlist}
+import spinal.core.internals.{Operator, DataAssignmentStatement}
 import spinal.core.sim.SimWorkspace
-import spinal.core.{BlackBox, Component, GlobalData, SpinalConfig, SpinalReport}
+import spinal.core.{BlackBox, Component, GlobalData, SpinalConfig, SpinalReport, SpinalError, SpinalProgress}
+import spinal.core.{noUndrivenCheck, cloneOf}
```

Comment:
I think noUndrivenCheck is not necessary, if you met this, you might miss to declare the input and output as in/out or master/slave.

Reply:
I don't see how input/output master/slave could impact this?
The functions `anyseq` `anyconst` `allseq` and `allconst` in VHDL are litterally only a signal with an attribute. No given value whatsoever and as such of course completely undriven.

Example:
```vhdl
architecture arch of FormalTestVHDL is
  attribute anyconst : boolean;
  signal toplevel_dut_1_io_a_anyconst : std_logic;
  attribute anyconst of toplevel_dut_1_io_a_anyconst : signal is true;
begin
  dut_1 : entity work.Dut
    port map ( 
      io_a => toplevel_dut_1_io_a_anyconst 
    );
  end process;
end arch;
```

Reply:
so does this equal to 
``` Verilog
assign dut_delay = $anyconst(delay_width);
```
If so the current syntax is working, then the check should be pass, so that the only modification might be in Emitter of VHDL, as I understand.

Reply:
Hmm I see. The problem is, that in VHDL the unconstrained variables are not an assignment at all and instead a simple attribute on an otherwise undriven signal. To convert the SpinalHDL assignment to the attribute I implemented a _pre processing_ phase `PhaseFormalVhdlFixups` in `FormalBootstraps.scala` which turns the assignment to an attribute.

Initially this conversion also removed the assignment itself from the statement graph. It could be left intact, and the implementation of VHDL could just ignore the statement. Something like so:

```diff
diff --git a/core/src/main/scala/spinal/core/formal/FormalBootstraps.scala b/core/src/main/scala/spinal/core/formal/FormalBootstraps.scala
index 22b719fba..a3bcddc94 100644
--- a/core/src/main/scala/spinal/core/formal/FormalBootstraps.scala
+++ b/core/src/main/scala/spinal/core/formal/FormalBootstraps.scala
@@ -307,10 +307,11 @@ override def impl(pc: PhaseContext): Unit = {
           val sig = cloneOf(target)
             .addAttribute(kind)
             .setCompositeName(target, kind)
-            .addTag(noUndrivenCheck)
+            // .addTag(noUndrivenCheck)
+          target.allowOverride()
           target := sig
         }
-        s.removeStatement()
+        // s.removeStatement()
       case _ =>
     }
   }
diff --git a/core/src/main/scala/spinal/core/internals/ComponentEmitterVhdl.scala b/core/src/main/scala/spinal/core/internals/ComponentEmitterVhdl.scala
index cad692584..3f8d8f42d 100644
--- a/core/src/main/scala/spinal/core/internals/ComponentEmitterVhdl.scala
+++ b/core/src/main/scala/spinal/core/internals/ComponentEmitterVhdl.scala
@@ -816,6 +816,7 @@ class ComponentEmitterVhdl(
 
   def emitAssignment(assignment: AssignmentStatement, tab: String, assignmentKind: String): String = {
     assignment match {
+      case r if r.source.isInstanceOf[Operator.Formal.RandomExp] => "" // handled in FormalBootstraps.PhaseFormalVhdlFixups
       case _ =>
         if (!assignment.target.isInstanceOf[SpinalStruct])
           s"$tab${emitAssignedExpression(assignment.target)} ${assignmentKind} ${emitExpression(assignment.source)};${emitLocation(assignment)}\n"
```

But now the `allowOverride` is required as Spinal thinks that the signal is assigned twice...

Reply:
So, if the expression can be identified as Formal.RandomExp then how about generate the attributes string for all Formal.RandomExp (Maybe it is in emitAttributesDef, I am not very familiar with the backends.) and skip it while emitAssignment, just like you do.